src/StreamJsonRpc/MessagePackFormatter.cs

```diff
@@ -2239,6 +2299,44 @@ public override bool TryGetArgumentByNameOrIndex(string? name, int position, Typ
                     this.formatter.deserializingMessageWithId = default;
                 }
```

Comment:
Go ahead and throw. The caller can make the choice to trace, or retry with a different type to deserialize as, etc.